### PR TITLE
Init RequestParser Implement

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.20)
```
Comment (john0312) [inline CMakeLists.txt]: As per the other PR.

File: src/Implement/RequestParser.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  "extended": true,
+  "results": [
+    {"value": 0, "type": "int64"},
+    {"value": 1.0e+3, "type": "decimal"}
+  ]
+}
+*/
+OperateData* RequestParser::ParseData(const std::string& request) {
+  OperateData* new_data = new OperateData();
+  new_data->header["method"] = request.substr(0, request.find(" "));
```
Comment (john0312) [inline src/Implement/RequestParser.cpp:36]: [Optional] Consider tokenizer for parsing

File: src/Implement/RequestParser.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  size_t url_key_start = request.find(" ") + 2;
+  size_t url_key_end = request.find(" ", request.find(" ") + 1);
+  new_data->header["url"] =
+      request.substr(url_key_start, url_key_end - url_key_start);
+  for (int i = 0; i < new_data->header["url"].length(); i++) {
```
Comment (john0312) [inline src/Implement/RequestParser.cpp:43]: Maybe add comments on what each segment is doing?

File: src/Implement/UnitTestRequestParser.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#include <gmock/gmock.h>
```
Comment (john0312) [inline src/Implement/UnitTestRequestParser.cpp:5]: Please add license header.

File: src/Implement/RequestParser.cpp (hunk 3 above)
Comment (younger15) [reply src/Implement/RequestParser.cpp:43]: Done

File: src/Implement/UnitTestRequestParser.cpp (hunk 4 above)
Comment (john0312) [reply src/Implement/UnitTestRequestParser.cpp:5]: This is not done yet.

**Review by john0312 — APPROVED** [review]
Please add license header before merging.

File: src/Implement/UnitTestRequestParser.cpp (hunk 4 above)
Comment (younger15) [reply src/Implement/UnitTestRequestParser.cpp:5]: Done

**Pull request merged**